### ISSUE TITLE
PYP last qn

### ISSUE THREAD

**Issue by danyelys** [nus-cs2030/2122-s2#594]
hi anyone knows why this wont compile? 
![Screenshot (162)](https://user-images.githubusercontent.com/100119338/165434708-137f52ed-9a63-42c6-9e85-99cb437f8ac4.png)

**Comment by mingshan2705** [comment]
I might be wrong but I think the "this" in line 24 refers to the MyStream<R> in line 22, and this.get() in line 22 will return an object of type R which causes an error since fn expects a type T object (?) maybe you can try replacing "this" with "MyStream.this" in line 22 and see if it compiles

**Comment by danyelys** [comment]
@mingshan2705 it works now with that. dint see that too HAHAH thank you!!

**Comment by danyelys** [comment]
@mingshan2705 how does MyStream.this know which MyStream it is pointing to

**Comment by Swagston20** [comment]
u need the MyStream or else it wil run recursively with the R get()

**Comment by mingshan2705** [comment]
> @mingshan2705 how does MyStream.this know which MyStream it is pointing to

Im not really sure also HAHA but something similar was discussed in recitation 7 qn 2.. maybe you can check that out

**Comment by shotnothing** [comment]
Yeah it doesn't really make logical sense as qualified this is a non-generic context, meaning its just something you need to remember that you can do. 

https://docs.oracle.com/javase/specs/jls/se10/html/jls-15.html#jls-15.8.4

**Comment by DikshantDulal** [comment]
This is the same issue as in recitation 07, q2b) as pointed out above. You run into a recursive loop when you this.get() because it's calling the get method of your global MyStream instead of the local MyStream which is what you want.
